awx/api/serializers.py

```diff
@@ -2201,6 +2201,89 @@ def create(self, validated_data):
         return return_data
 
 
+class BulkHostDeleteSerializer(serializers.Serializer):
+    hosts = serializers.ListField(
+        allow_empty=False,
+        max_length=250,
+        write_only=True,
+        help_text=_('List of hosts ids to be deleted, e.g. [105, 130, 131, 200]'),
+    )
+
+    class Meta:
+        model = Host
+        fields = 'hosts'
+        read_only_fields = ()
+
+    def validate(self, attrs):
+        request = self.context.get('request', None)
+
+        # Validating the number of hosts to be deleted
+        if len(attrs['hosts']) > settings.BULK_HOST_MAX_DELETE:
+            raise serializers.ValidationError(
+                {
+                    "ERROR": 'Number of hosts exceeds system setting BULK_HOST_MAX_DELETE',
+                    "BULK_HOST_MAX_DELETE": settings.BULK_HOST_MAX_DELETE,
+                    "Hosts_count": len(attrs['hosts']),
+                }
+            )
+
+        # Getting list of all host objects, filtered by the list of the hosts to delete
+        attrs['in_inv'] = Host.objects.get_queryset().filter(pk__in=attrs['hosts'])
+
+        # Converting the queryset data in a dict. to reduce the number of queries when
+        # manipulating the data
+        hosts_data = attrs['in_inv'].values()
```

Comment:
you later call `values` again in `delete` perhaps instead you could do
attrs['host_data'] = attrs['in_inv'].values() and use it again later in `delete` method